### ISSUE TITLE
Uncaught TypeError: Cannot read property 'isEqual' of undefined

### ISSUE THREAD

**Issue by jjlorenzo** [atom/atom#16394]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.24.0-beta0 x64
**Electron**: 1.6.15
**OS**: Mac OS X 10.12.6
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'isEqual' of undefined

```
At /Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-registry.js:233

TypeError: Cannot read property 'isEqual' of undefined
    at editorsWithMaintainedConfig.forEach (/Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-registry.js:233:56)
    at Set.forEach (native)
    at subscriptions.add.config.onDidChange (/Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-registry.js:232:52)
    at /Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/src/config.js:955:26
    at Function.module.exports.Emitter.simpleDispatch (/Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at Config.module.exports.Config.emitChangeEvent (/Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/src/config.js:884:35)
    at Config.module.exports.Config.setRawValue (/Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/src/config.js:670:25)
    at Config.module.exports.Config.set (/Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/src/config.js:248:20)
    at SettingsPanel.set (/Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/node_modules/settings-view/lib/settings-panel.js:237:27)
    at HTMLInputElement.changeHandler (/Users/jjlorenzo/Applications/Atom Beta.app/Contents/Resources/app/node_modules/settings-view/lib/settings-panel.js:174:25)
```

### Commands

```
  7x -2:36.6.0 core:move-right (input.hidden-input)
     -2:36 editor:move-to-end-of-screen-line (input.hidden-input)
  2x -2:35.7.0 editor:move-to-first-character-of-line (input.hidden-input)
  3x -2:35.1.0 core:move-down (input.hidden-input)
     -1:46.5.0 core:copy (input.hidden-input)
     -1:44.7.0 core:paste (input.hidden-input)
     -1:43.7.0 core:move-down (input.hidden-input)
  2x -1:43.5.0 editor:move-to-first-character-of-line (input.hidden-input)
     -1:42.9.0 editor:newline (input.hidden-input)
     -1:42.4.0 core:save (input.hidden-input)
     -1:33.2.0 settings-view:open (input.hidden-input)
     -0:42.9.0 nuclide-settings:toggle (input.hidden-input)
 23x -0:37.4.0 core:backspace (input.hidden-input)
     -0:28.9.0 core:close (input.hidden-input)
     -0:25.4.0 settings-view:open (input.hidden-input)
     -0:16.3.0 core:confirm (input.hidden-input)
```

### Non-Core Packages

```
file-icons 2.1.15 
highlight-selected 0.13.1 
language-babel 2.81.0 
language-graphql 0.9.0 
language-haskell 1.16.2 
language-ini 1.19.0 
language-lua 0.9.11 
language-ocaml 1.9.5 
language-rust 0.4.12 
language-scala 1.1.9 
language-swift 0.5.0 
language-thrift 1.0.2 
no-title-bar 1.11.3 
nuclide 0.271.0 
nuclide-format-js 0.0.42 
set-syntax 0.4.0 
sort-lines 0.18.0 
tool-bar 1.1.0 
```

**Comment by rsese** [comment]
Thanks for the report!  Can you confirm a few things for us?

* what were you doing at the time of the error?
* is the error is reproducible?
* if reproducible, do you see the error in safe mode (atom --safe)?

**Comment by robwalton** [comment]
I also see this error.

* I get this error when toggling the Settings->Editor->"Show Line Number" checkbox. The checkmark toggles, but seems not to effect the line numbers in the gutter.
* The error is reproducible. It occurs every time I try this.
* It *does not* occur when started in safe mode.

Switching to safe mode let me toggle the setting with desired effect. So I'm happy at least!

**Comment by Aerijo** [comment]
@robwalton The best way to handle it in that case would be to selectively enable / disable packages until you find which is causing the problem. Then raise an issue on the respective repo (if not already raised).

**Comment by robwalton** [comment]
It turns out that toggling many of the checkboxes near "Show Line Number" also causes it. It did not do this before I installed a bunch of plugins. When I have time, I will work out which is causing it. Note also that I'm **not** using a beta version like the original reporter. I attach my error report info.
[atom_error_report.txt](https://github.com/atom/atom/files/1731604/atom_error_report.txt)

**Comment by robwalton** [comment]
@Aerijo. Thanks for your help. So I've disabled and reenabled my plugins and of course now it works fine!

A dodgy community package should not be able to break so much of the core editor code though (it turns out almost all the settings resulted in the same error). If it happens again I'll check out the atom core source and see if I can at least improve the error message to give a better indication of cause.

**Comment by 50Wliu** [comment]
@robwalton since this appears to be an issue with a community package, I'm going to close this.  However, please feel free to open a new issue that requests for this error message to be improved, or to create a PR.

**Comment by darkn3rd** [comment]
I just rain into this on 1.27.0 x64 by clicking on "show invisibles".

**Comment by TheMengzor** [comment]
Disabled todo-show plugin and everything works fine

**Comment by evanperryg** [comment]
Looks like it may be an issue with the todo-show plugin, per TheMengzor's comment. The exception occurred when I tried to modify the "Preferred Line Length" setting.

I tried disabling todo-show and the exception still occurred, but after restarting atom (leaving todo-show disabled) I was able to change the preferred line length without a problem. Re-enabling todo-show after changing the setting does not seem to have any damaging effect.
